client/blocks/login/index.jsx

```diff
@@ -40,12 +40,30 @@ class Login extends Component {
 	};
 
 	componentWillMount = () => {
-		if ( ! this.props.twoStepNonce && this.props.twoFactorAuthType && typeof window !== 'undefined' ) {
+		if ( typeof window === 'undefined' ) {
+			return;
+		}
+
+		if ( ! this.props.twoStepNonce && this.props.twoFactorAuthType ) {
 			// Disallow access to the 2FA pages unless the user has received a nonce
 			page( login( { isNative: true } ) );
 		}
 	};
 
+	componentWillReceiveProps = ( nextProps ) => {
+		if ( typeof window === 'undefined' ) {
```

Comment:
`componentWillReceiveProps` is never called on the server, so we can skip check against `window`.